test/unit/CodeGenerator/Filesystem/FilePutContentsFileWriterTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Unit\CodeGenerator\Filesystem;
+
+use OnMoon\OpenApiServerBundle\CodeGenerator\Filesystem\FilePutContentsFileWriter;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+
+use function Safe\unlink;
+
+use const DIRECTORY_SEPARATOR;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\Test\Unit\CodeGenerator\Filesystem\FilePutContentsFileWriter
+ */
+final class FilePutContentsFileWriterTest extends TestCase
+{
+    public function testWriteCreatesFile(): void
+    {
+        $path     = 'test/';
+        $filename = 'testFilename.txt';
+        $fullPath = $path . DIRECTORY_SEPARATOR . $filename;
+
+        $fileWriter = new FilePutContentsFileWriter(0755);
+
+        Assert::assertFileDoesNotExist($fullPath);
+        $fileWriter->write($path, $filename, 'SomeContents');
+        Assert::assertFileExists($fullPath);
```

Comment:
https://coveralls.io/builds/32505321/source?filename=src/CodeGenerator/Filesystem/FilePutContentsFileWriter.php

These things need to be verified too:
- The contents of the file created
- That the directory structure is created if we pass a path containing directories that do not exist
- That the created directories have the expected permissions